k8s/upgrades/0.5.x-0.6.0/oebs_update.sh

```diff
@@ -0,0 +1,113 @@
+kubectl patch deployment --namespace $ns --type json $r_dep -p "$(cat patch-strategy-recreate.json)"
+rc=$?; if [ $rc -ne 0 ]; then echo "ERROR: $rc"; exit; fi
+
+echo "Patching Controller Deployment upgrade strategy as recreate"
+kubectl patch deployment --namespace $ns --type json $c_dep -p "$(cat patch-strategy-recreate.json)"
+rc=$?; if [ $rc -ne 0 ]; then echo "ERROR: $rc"; exit; fi
+
+c_rs=$(kubectl get rs -o name --namespace $ns | grep $c_dep | cut -d '/' -f 2)
+r_rs=$(kubectl get rs -o name --namespace $ns | grep $r_dep | cut -d '/' -f 2)
+
+################################################################ 
+# STEP: Update patch files with appropriate container names    #
+#                                                              # 
+# NOTES: Placeholder "pvc-<deploy-hash>-ctrl/rep-con in the    #
+# patch files are replaced with container names derived from   #
+# the PV in the previous step                                  #  
+################################################################
+
+sed "s/@pvc-name[^ \"]*/$pvc/g" replica.patch.tpl.yml > replica.patch.tpl.yml.0
+sed "s/@replica_node_label[^ \"]*/$replica_node_label/g" replica.patch.tpl.yml.0 > replica.patch.tpl.yml.1
+sed "s/@r_name[^ \"]*/$r_name/g" replica.patch.tpl.yml.1 > replica.patch.yml
+
+sed "s/@pvc-name[^ \"]*/$pvc/g" controller.patch.tpl.yml > controller.patch.tpl.yml.0
+sed "s/@c_name[^ \"]*/$c_name/g" controller.patch.tpl.yml.0 > controller.patch.tpl.yml.1
+sed "s/@rep_count[^ \"]*/$rep_count/g" controller.patch.tpl.yml.1 > controller.patch.yml
+
+################################################################
+# STEP: Patch OpenEBS volume deployments (controller, replica) #  
+#                                                              #
+# NOTES: Strategic merge patch is used to update the volume w/ #  
+# rollout status verification                                  #  
+################################################################
+
+# PATCH JIVA REPLICA DEPLOYMENT ####
+echo "Upgrading Replica Deployment to 0.6"
+kubectl patch deployment --namespace $ns $r_dep -p "$(cat replica.patch.yml)"
+rc=$?; if [ $rc -ne 0 ]; then echo "ERROR: $rc"; exit; fi
+
+rollout_status=$(kubectl rollout status --namespace $ns deployment/$r_dep)
+rc=$?; if [[ ($rc -ne 0) || !($rollout_status =~ "successfully rolled out") ]];
+then echo "ERROR: $rc"; exit; fi
+
+#### PATCH CONTROLLER DEPLOYMENT ####
+echo "Upgrading Controller Deployment to 0.6"
+kubectl patch deployment  --namespace $ns $c_dep -p "$(cat controller.patch.yml)"
+rc=$?; if [ $rc -ne 0 ]; then echo "ERROR: $rc"; exit; fi
+
+rollout_status=$(kubectl rollout status --namespace $ns  deployment/$c_dep)
+rc=$?; if [[ ($rc -ne 0) || !($rollout_status =~ "successfully rolled out") ]];
+then echo "ERROR: $rc"; exit; fi
+
+################################################################
+# STEP: Remove Stale Controller Replicaset                     #
```

Comment:
I guess we are not trying to rollback in case of failure. How about restoring to a particular deployment version in case of failures?

Reply:
automatic rollback via this script is not provided. Added that as a limitation.